src/_pytest/mark/structures.py

```diff
@@ -284,27 +285,48 @@ def test_function():
 
     _config = None
 
+    # These marks are always treated as known for the warning, which is designed
```

Comment:
I suppose the proper way to handle this is to make each plugin to register their known marks, just like any plugin (for example `skipping.py` should register `skip`, `skipif` and `xfail` markers).

Reply:
:man_facepalming: They already do, actually.  That'll simplify the implementation a bit...